hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/HoodieDeltaStreamer.java

```diff
@@ -684,6 +691,7 @@ public DeltaSync getDeltaSync() {
           }
         } finally {
           shutdownAsyncServices(error);
+          executor.shutdownNow();
```

Comment:
Hi ethan, we could use `shutdownAsyncServices(error)`to close async compaction and async clustering properly, and quit the loop and finished future directly. Is it necessary to still call `executor.shutdownNow();`here?

Reply:
I tried to remove it, but then the DeltaStreamer Spark app would not exit.  This is to shut down the DeltaSync service's executor thread pool.